server/server.js

```diff
@@ -0,0 +1,94 @@
+const express = require('express');
+const cors = require('cors');
+const port = 3000;
+
+const server = express();
+server.use(express.json());
+server.use(cors());
+
+const sendUserError = (msg, res) => {
+  res.status(422);
+  res.json({ Error: msg });
+  return;
+};
+
+let notes = [
+  {
+    id: 0,
+    "tags": ["tag", "otherTag"],
+    "title": "Note Title",
+    "textBody": "Note Body",
+  }
+];
+server.get('/note/get/all', (req, res) => {
+  res.json(notes);
+});
+
+server.get('/note/get/:id', (req, res) => {
+  const { id } = req.params;
+  const findNoteById = note => {
+    if (note.id === id) return note;
+  };
+  const foundNote = notes.find(findNoteById);
+  if (!foundNote) {
+    return sendUserError('No Note found by that ID', res);
+  } else {
+    res.json(foundNote);
+  }
+});
+
+server.post('/note/create', (req, res) => {
+  const { tags, title, textBody } = req.body;
+  let noteId
+
+  if (notes.length === 0) {
+    noteId = 0
+  } else {
+    noteId = notes[notes.length - 1].id + 1
+  }
+
+  const newNote = { tags, title, textBody, id: noteId };
+  if ( !title || !textBody ) {
+    return sendUserError(
+      'Error creating note.',
+      res
+    );
+  }
+
+  notes.push(newNote);
+  res.json(notes);
+});
+
+server.put('/note/edit/:id', (req, res) => {
+  const { id } = req.params;
+  const { tags, title, textBody } = req.body;
+  const findNoteById = note => {
+    return note.id === id;
+  };
+  const foundNote = notes.find(findNoteById);
+  if (!foundNote) {
+    return sendUserError('No Note found by that ID', res);
+  } else {
+    if (tags.length > 0) foundNote.tags = tags;
+    if (title) foundNote.title = title;
+    if (textBody) foundNote.textBody = textBody;
+    res.json(notes);
+  }
+});
+
+server.delete('/note/delete/:id', (req, res) => {
+  const { id } = req.params;
+  const foundNote = notes.find(note => note.id === id);
+
+  if (foundNote) {
+    notes = notes.filter(note => note.id !== id);
+    res.status(200).json(notes);
+  } else {
+    sendUserError('No note by that ID exists in the note DB', res);
+  }
+});
+
+server.listen(port, err => {
+  if (err) console.log(err);
+  console.log(`server is listening on port ${port}`);
+});
```

Comment:
## Trello Set Up:
 * [x] Create a Trello account
 * [x] Create a new board called "Lambda Notes - {Your Name}"
 * [x] Create lists titled backlog,To Do, In Progress, and Done
 * [x] Fill in the To Do list with the MVP features listed below
 * [x] Fill in the backlog list with all the extra features listed below
 * [x]  Add your Trello URL to your project's README.md file. Commit the change, push it to your repository & submit a pull request
### MVP Features:
 * [x] Display a list of notes
-- Watch your sizes and spacings. I'm viewing your project on my MacBook Pro and everything is super small. The height of the webpage also only comes down to about halfway down the screen. I assume this is because height is set to auto or a min-height - try setting the vh to 100  on the main container to see if that changes anything. 
-- You were given a design file - as a developer trying to fulfill clients wishes, try to make it as pixel perfect as possible and go with the stylings that they have asked for. This includes removing text-decoration and the default font-color when making components a link. 
 * [x] Create a note with a title and content
-- Creating the note works, but the `save` button should link back to the list of notes as opposed to refreshing the page with empty input boxes. What do you have to do to change the routing there? 
 * [x] View an existing note
-- Viewing a singular note works. Again, watch the spacing to match the design file as close as possible. 
 * [x] Edit an existing note
-- Pass state through the link so that you have access to the note's title and content in the edit form. You need to be able to have those values already in the input box so that it can indeed be edited...the way it is now, you would have to remember what your note's title and content would be in order to edit it properly. 
 * [x] Delete an existing note
-- Works. 
 * [x] Wire up your static React App to our Notes API.

Overall, MVP has been achieved with these few caveats...because this is a front-end project, more attention needs to be placed on the design and how things appear on the page. 

The logic that needs to be fixed is very minor. You're able to pass state through the Link component - Tyler McGinnis has a great YouTube video on how to do that. The Create component needs to link back to the list of notes after it has been saved. 

Great job! Thanks for linking your Trello board so I can see that as well.

Reply:
It looks like you got off to a good start. I was able to run your application and execute all CRUD endpoints. Although your application is not 100% pixel perfect, the overall styling looks good. However, I will like you to improve and fix the following things in order to meet MVP requirements:

1. Work on properly nesting your code statements and writing comments to help other developers understand your code.

2. You have a console error. The first error " Invalid value for prop `component` on <a> tag. Either remove it from the element, or pass a string or number value to keep it in the DOM."
The second error is "Failed prop type: Invalid prop `centered` of type `string` supplied to `Modal`, expected `boolean".

3. Your application should be the full width and full height of the browser window. Having extra white space on the left and right side of your application does not look very good. To fix this, you can use view width and view height in your CSS (e.g min-height: 100vh, width: 80vw). 

4. Your application should feel like a single application. Navigating to the note view component  causes the application to reload when redirecting. This is because you are using ```<a>``` tags instead of ```<Link>``` tags that can be imported from react-router-dom. Also, using ```<a>``` tags does not allow a component to receive react router props.

5. The values of the input fields in the note edit component should be set to the note title and content.